src/microbe_stage/Microbe.cs

```diff
@@ -1957,6 +1988,36 @@ private void ProcessPhysicsForEngulfing()
         }
     }
 
+    private void HandleSprint(float delta)
+    {
+        if (isSprinting)
+        {
+            // Drain atp
+            var cost = BaseMovementCost(delta) * (1.0f + stress);
+            if (Compounds.GetCompoundAmount(atp) < Compounds.Capacity * Constants.MIN_ATP_TO_CONTINUE_SPRINT)
+            {
+                isSprinting = false;
+            }
+            else
+            {
+                Compounds.TakeCompound(atp, cost);
```

Comment:
I think this should be rewritten to try to take the compound amount first and then check if enough could be taken, otherwise disable sprint.